### PR TITLE
add focal loss

### PR DESCRIPTION
**Pull request by lucamarini22**
add focal loss

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -1,42 +1,65 @@
...
+            args.snapshot_path = os.path.join(
+                args.results_dir, f"unet-model-{seed}-{today_str}-{args.loss}.pth.tar"
+            )
             args.tensorboard_logdir = os.path.join(args.tensorboard, f"model_{seed}/")
             os.makedirs(args.results_dir, exist_ok=True)
             os.makedirs(args.tensorboard, exist_ok=True)
+            args.today = today_str
             train(args)
     elif args.mode == "predict":
-        raise NotImplementedError("here all images should be predicted in a loop using the models trained before")
```
Comment (GabrieleMeoni) [inline src/main.py:15]: This should be fixed in a future release. Please, open a in issue for it.

File: src/main.py [hunk 2]
```diff
@@ -1,42 +1,65 @@
...
-    parser.add_argument('--pos-weight', type=float, default=1, help="positional weight for the floating object class, large values counteract")
+    parser.add_argument("--data-path", type=str, default="/data")
+    parser.add_argument("--batch-size", type=int, default=8)
+    parser.add_argument("--workers", type=int, default=0)
+    parser.add_argument("--image-size", type=int, default=128)
+    parser.add_argument("--device", type=str, choices=["cpu", "cuda"], default="cuda")
+    parser.add_argument("--epochs", type=int, default=50)
+    parser.add_argument("--learning-rate", type=float, default=1e-3)
+    parser.add_argument(
+        "--augmentation-intensity",
```
Comment (GabrieleMeoni) [inline src/main.py:40]: What kind of augmentation is used here? Please, investigates.

**Review by GabrieleMeoni — CHANGES_REQUESTED** [review]

File: src/main.py (hunk 1 above)
Comment (lucamarini22) [reply src/main.py:15]: done

File: src/main.py (hunk 2 above)
Comment (lucamarini22) [reply src/main.py:40]: true, it is not so clear to me right now, should I also open an issue for this?

Comment (GabrieleMeoni) [reply src/main.py:40]: Then, investigate it as a new step.

**Review by GabrieleMeoni — APPROVED** [review]

**Pull request merged**